numpy/core/src/multiarray/array_method.h

```diff
@@ -224,12 +224,13 @@ HPyType_LEGACY_HELPERS(PyArrayMethodObject)
  * on the `ArrayMethod` itself.
  */
 typedef struct {
+    PyObject_HEAD
     HPyField *dtypes; /* PyArray_DTypeMeta **dtypes */
     HPyField method; /* PyArrayMethodObject *method */
     int nargs; /* method->nin + method->nout */
 } PyBoundArrayMethodObject;
 
-HPyType_HELPERS(PyBoundArrayMethodObject)
+HPyType_LEGACY_HELPERS(PyBoundArrayMethodObject)
```

Comment:
I guess this had to become a legacy type so that someone could call `HPy_AsPyObject` on it?

Reply:
🤔 good point. I don't think we ever specified that.
It is not required on CPython because any object is a `PyObject` anyway. So, it is always possible to convert.
It is not required for GraalPy because we would return a _wrapper_ that emulates the `PyObject`.
Is it required for PyPy? If so, we should discuss that and maybe properly specify (and enforce spec in debug mode).

Reply:
PyPy works around it but it would be more efficient if the HELPER was correct.